runtime/lua/vim/filetype.lua

```diff
@@ -1007,6 +1034,150 @@ local extension = {
+      return 'dsl'
+    end
+    return 'structurizr'
+  end,
+  edf = 'edif',
+  edfi = 'edif',
+  edo = 'edif',
+  edn = function(path, bufnr)
+    require('vim.filetype.detect').smi(bufnr)
+  end,
+  smil = function(path, bufnr)
+    if getline(bufnr, 1):find('<%?%s*xml.*%?>') then
+      return 'xml'
+    end
+    return 'smil'
+  end,
+  smi = function(path, bufnr)
+    require('vim.filetype.detect').smi(bufnr)
+  end,
+  install = function(path, bufnr)
+    if getline(bufnr, 1):lower():find('<%?php') then
+      return 'php'
+    end
+    vim.fn['dist#ft#SetFileTypeSH']('bash')
+  end,
+  pm = function(path, bufnr)
+    local line = getline(bufnr, 1)
+    if line:find('XPM2') then
+      return 'xpm2'
+    elseif line:find('XPM') then
+      return 'xpm'
+    else
+      return 'perl'
+    end
+  end,
+  me = function(path, bufnr)
+    local filename = vim.fn.fnamemodify(path, ':t'):lower()
+    if filename ~= 'read.me' and filename ~= 'click.me' then
+      return 'nroff'
+    end
+  end,
+  reg = function(path, bufnr)
+    local line = getline(bufnr, 1):lower()
+    if line:find('^regedit[0-9]*%s*$') or line:find('^windows registry editor version %d*%.%d*%s*$') then
+      return 'registry'
+    end
+  end,
+  decl = function(path, bufnr)
+    require('vim.filetype.detect').decl(bufnr)
+  end,
+  dec = function(path, bufnr)
+    require('vim.filetype.detect').decl(bufnr)
+  end,
+  dcl = function(path, bufnr)
+    require('vim.filetype.detect').dcl(bufnr)
+  end,
+  web = function(path, bufnr)
+    require('vim.filetype.detect').web(bufnr)
+  end,
+  ttl = function(path, bufnr)
+    local line = getline(bufnr, 1)
+    if line:find('^@?prefix') or line:find('^@?base') then
+      return 'turtle'
+    end
+    return 'teraterm'
+  end,
+  am = function(path, bufnr)
+    local filename = vim.fn.fnamemodify(path, ':t'):lower()
+    if not filename:find('makefile%.am$') then
+      return 'elf'
+    end
+  end,
+  ['m4'] = function(path, bufnr)
+    local filename = vim.fn.fnamemodify(path, ':t'):lower()
+    if not filename:find('html%.m4$') and not filename:find('fvwm2rc') then
+      return 'm4'
+    end
+  end,
+  hw = function(path, bufnr)
+    if getline(bufnr, 1):lower():find('<%?php') then
+      return 'php'
+    end
+    return 'virata'
+  end,
+  module = function(path, bufnr)
+    if getline(bufnr, 1):lower():find('<%?php') then
+      return 'php'
+    end
+    return 'virata'
+  end,
+  pkg = function(path, bufnr)
+    if getline(bufnr, 1):find('<%?php') then
+      return 'php'
+    end
+    return 'virata'
+  end,
+  ms = function(path, bufnr)
+    if not require('vim.filetype.detect').nroff(bufnr) then
+      return 'xmath'
+    end
+  end,
+  t = function(path, bufnr)
+    if not require('vim.filetype.detect').nroff(bufnr) and not require('vim.filetype.detect').perl(path, bufnr) then
+      return 'tads'
+    end
+  end,
+  class = function(path, bufnr)
+    if not getline(bufnr, 1):lower():find('^%xca%xfe%xba%xbe') then
```

Comment:
This should be checking for the actual bytes `\xca\xfe\xba\xbe`

```suggestion
    if not getline(bufnr, 1):find('^\202\254\186\190') then
```